### PR TITLE
Add scipy and numpy nightly wheels to dev dependencies

### PR DESCRIPTION
**Pull request by WilliamJamieson**
This PR adds the scipy and numpy nightly wheels to the dev dependencies, so that we can detect issues like the scipy 1.10.0 release prior to that release.

**Checklist for maintainers**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [ ] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by zacharyburnett** [comment]
It looks like the versions of `numpy` and `scipy` are not affected in the environment:
https://github.com/spacetelescope/jwst/actions/runs/3849362407/jobs/6558268913
```ini
numpy==1.24.1
...
scipy==1.9.3
```

**Comment by braingram** [comment]
Does the tox.ini file need PIP_EXTRA_INDEX_URL like in asdf:
https://github.com/asdf-format/asdf/blob/f5cc1aa53e663fcae70bbb7127c0433ce6f9644f/tox.ini#L6-L7
```ini
setenv =
    devdeps: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/scipy-wheels-nightly/simple
```

**Comment by WilliamJamieson** [comment]
> Does the tox.ini file need PIP_EXTRA_INDEX_URL like in asdf: [asdf-format/asdf@`f5cc1aa`/tox.ini#L6-L7](https://github.com/asdf-format/asdf/blob/f5cc1aa53e663fcae70bbb7127c0433ce6f9644f/tox.ini#L6-L7)
> 
> ```ini
> setenv =
>     devdeps: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/scipy-wheels-nightly/simple
> ```

Yes it probably does

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -66,6 +66,12 @@ jobs:
           - toxenv: test-opencv-xdist
             os: ubuntu-latest
             python-version: '3.10'
+          - toxenv: test-devdeps-xdist
+            os: ubuntu-latest
+            python-version: '3.10'
+          - toxenv: test-devdeps-xdist
+            os: macos-latest
+            python-version: '3.10'
```
Comment (jdavies-st) [inline .github/workflows/ci.yml:74]: Probably not needed to run devdeps jobs for every pull request in ci.yml.  There's a ci_cron.yml workflow here that runs the devdeps jobs every Monday morning that is probably more appropriate for this.  And feel free to run it more often if weekly is not enough to catch these issues.

Comment (WilliamJamieson) [reply .github/workflows/ci.yml:74]: This is up to @zacharyburnett, as they added those changes. My original intent was to just test these as part of our weekly regression tests.

Comment (zacharyburnett) [inline .github/workflows/ci.yml:74]:
```suggestion
```

good point, we can run them in the weekly job

**Comment by nden** [comment]
I think this is superseded by #7581. If still relevant, please reopen and update.

**Pull request closed without merging**